### PR TITLE
docs: add global variable rule disabling FAQ for ESLint

### LINKED ISSUE #9582: Docs: globalThis without ignores of no-var and no-namespace
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

A `typescript-eslint`-blessed solution for the problem of declaring `globalThis` properties without ignores:

```ts
declare global {
  var postgresSqlClient: Sql; // 💥 Unexpected var, use let or const instead. eslint (no-var)
}
```

`declare module globalThis` option:

```ts
declare module globalThis {
  let postgresSqlClient: Sql;
} // 💥 ES2015 module syntax is preferred over namespaces. eslint (@typescript-eslint/no-namespace)
```

### Affected URL(s)

- https://v8--typescript-eslint.netlify.app/rules/no-namespace/

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

- [x] Addresses an existing open issue: fixes #9582
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Adds similar examples to what's in the docs issue, along with suggested remediations, tips on disable comments, and a link to the two issues.

💖

### CODE REVIEW

**Comment by Josh-Cena** [comment]
1. This should be in the FAQ of the individual rules since there are only two involved.
2. I still think a configuration is the right way to go, especially `no-var` since you should absolutely _not_ have global `declare let`, but... sure, let's have FAQ in the interim.

**Comment by JoshuaKGoldberg** [comment]
> of the individual rules

We don't have a file for `no-var` yet though 😕 . I can add one to `no-namespace.mdx` on its own, but I think t's a lot more common to `declare global` + `var`...

**Review by bradzacher — APPROVED** [review]

**Review by Josh-Cena — DISMISSED** [review]

**Review by kirkwaiblinger — DISMISSED** [review]

**Review by kirkwaiblinger — APPROVED** [review]

File: packages/eslint-plugin/docs/rules/no-namespace.mdx [hunk 1]
```diff
@@ -137,6 +137,9 @@ You might consider using [ESLint disable comments](https://eslint.org/docs/lates
 
 ## Further Reading
 
+{/* cspell:disable-next-line */}
```
Comment (Josh-Cena) [inline packages/eslint-plugin/docs/rules/no-namespace.mdx:140]: FYI I think you can configure cspell to ignore anchors using ignore patterns

Comment (JoshuaKGoldberg) [reply packages/eslint-plugin/docs/rules/no-namespace.mdx:140]: oh?? OH?? That would be so nice...

I can't find anything Googling, but if you have a reference I would _very very happily_ send a PR to enable it & remove any now-unnecessary disables/ignores.

**Pull request merged**